merlin/models/tf/blocks/core/transformations.py

```diff
@@ -37,6 +37,12 @@ def call(self, inputs: TabularData, **kwargs) -> TabularData:
             if isinstance(val, tuple):
                 values = val[0][:, 0]
                 row_lengths = val[1][:, 0]
+
+                if values.dtype == tf.float32:
+                    values = tf.cast(values, tf.int32)
+                if row_lengths.dtype == tf.float32:
```

Comment:
I am not sure if when mixed precision is enabled (like below) that would after the dtypes here to tf.float16.
In any case, it would be safer to check if the number is a float by using `row_lengths.dtype.is_floating`

```
policy = mixed_precision.Policy('mixed_float16')
mixed_precision.set_global_policy(policy)
```